### PR TITLE
Rename Decimal to Float

### PR DESCRIPTION
**Pull request by hubertp**
### Pull Request Description

Implements #6889.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.

### CODE REVIEW

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data/Type/Enso_Types.enso [hunk 1]
```diff
@@ -18,7 +18,7 @@ polyglot java import org.enso.base.polyglot.NumericConverter
 most_specific_value_type : Any -> Boolean -> Value_Type
 most_specific_value_type value use_smallest=False =
     case value of
-        _ : Decimal     -> Value_Type.Float Bits.Bits_64
+        _ : Float     -> Value_Type.Float Bits.Bits_64
```
Comment (radeusgd) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Data/Type/Enso_Types.enso:21]: So good to see the Enso types and Value_Types finally consistent 😃

**Review by radeusgd — APPROVED** [review]
Looks great!

**Review by GregoryTravis — APPROVED** [review]

File: engine/runtime/src/bench/java/org/enso/interpreter/bench/benchmarks/semantic/TypePatternBenchmarks.java [hunk 2]
```diff
@@ -88,7 +88,7 @@ public void matchOverAny(Blackhole matter) {
   }
 
   @Benchmark
-  public void matchOverDecimal(Blackhole matter) {
+  public void matchOverFloat(Blackhole matter) {
```
Comment (JaroslavTulach) [inline engine/runtime/src/bench/java/org/enso/interpreter/bench/benchmarks/semantic/TypePatternBenchmarks.java]: Changing name of a `@Benchmark` will disconnect it from the history of previous measurements and [affect the chart](https://enso-org.github.io/engine-benchmark-results/engine-benchs.html).

File: engine/runtime/src/main/java/org/enso/interpreter/runtime/builtin/Number.java [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
+import org.enso.interpreter.node.expression.builtin.number.Float;
 import org.enso.interpreter.node.expression.builtin.number.Integer;
 import org.enso.interpreter.runtime.data.Type;
 
 /** A container for all number-related builtins. */
 public class Number {
   private final Builtin integer;
   private final Builtin number;
-  private final Builtin decimal;
+  private final Builtin ensoFloat;
```
Comment (JaroslavTulach) [inline engine/runtime/src/main/java/org/enso/interpreter/runtime/builtin/Number.java:12]: I see. `float` is a keyword.

**Review by JaroslavTulach — APPROVED** [review]
Changes in Java & Scala code look sane and I am surprise there isn't even too much of them. If the tests run, merge it in!

**Review by farmaazon — APPROVED** [review]
Accepting Tiny GUI changes

File: engine/runtime/src/main/java/org/enso/interpreter/runtime/builtin/Number.java (hunk 3 above)
Comment (hubertp) [reply engine/runtime/src/main/java/org/enso/interpreter/runtime/builtin/Number.java:12]: Yeap

File: engine/runtime/src/bench/java/org/enso/interpreter/bench/benchmarks/semantic/TypePatternBenchmarks.java (hunk 2 above)
Comment (hubertp) [reply engine/runtime/src/bench/java/org/enso/interpreter/bench/benchmarks/semantic/TypePatternBenchmarks.java]: Reverted with a comment

Comment (Akirathan) [reply engine/runtime/src/bench/java/org/enso/interpreter/bench/benchmarks/semantic/TypePatternBenchmarks.java]: Another use-case for *versioning* mentioned in https://github.com/enso-org/enso/issues/5714.

**Pull request merged**